tests/pytorch/test_parallel_cross_entropy.py

```diff
@@ -61,19 +62,33 @@ def one_iteration_test(
         test_loss = self.test_loss_func(
             self.input_test, self.tar_test, label_smoothing, reduce_loss, None
         )
-        if reduce_loss:
-            test_loss.backward()
 
         ref_loss = self.ref_loss_func(self.input_ref, self.tar_ref)
-        if reduce_loss:
+
+        # Handle backward pass based on the test scenario
+        if use_sum_backward:
+            # Special case: use sum().backward() to trigger gradient scaling bug
+            # This should only be used with reduce_loss=False
+            assert not reduce_loss, "use_sum_backward should only be used with reduce_loss=False"
+            test_loss.sum().backward()
+            ref_loss.sum().backward()
+        elif reduce_loss:
+            test_loss.backward()
             ref_loss.backward()
 
-        test_loss = torch.flatten(test_loss) if not reduce_loss else test_loss
+        # Compare losses
+        if use_sum_backward:
+            # When using sum().backward(), compare the summed losses
+            torch.testing.assert_close(test_loss.sum(), ref_loss.sum(), check_dtype=False)
+        else:
+            test_loss = torch.flatten(test_loss) if not reduce_loss else test_loss
+            torch.testing.assert_close(test_loss, ref_loss, check_dtype=False)
```

Comment:
We still want to check that each entry of the loss output is correct, even if we're going to sum afterward:
```suggestion
        test_loss = torch.flatten(test_loss) if not reduce_loss else test_loss
```